fsm/interceptors.go

```diff
@@ -429,3 +429,52 @@ func (s *StartCancelPair) decisionsContainStartCancel(in []*swf.Decision, cancel
 	}
 	return -1
 }
+
+func CloseDecisionTypes() []string {
+	return []string{
+		swf.DecisionTypeCompleteWorkflowExecution,
+		swf.DecisionTypeCancelWorkflowExecution,
+		swf.DecisionTypeFailWorkflowExecution,
+
+		// swf.DecisionTypeContinueAsNewWorkflowExecution is technically a close type,
+		// but swfsm currently doesn't really treat it as one and there's
+		// a lot of special logic, so excluding it from here for now
+	}
+}
+
+func CloseDecisionIncompatableDecisionTypes() []string {
+	return []string{
+		// TODO: flush out this list with other incompatible types
+		// TODO: see https://console.aws.amazon.com/support/home?#/case/?caseId=5223303261&displayId=5223303261&language=en
```

Comment:
is there any more `TODO` here or can it be removed?

Reply:
I'm still waiting for AWS to respond this. There's no docs, so I'm guessing they're having to work with the dev team to get this info from the code. I'm thinking we should merge this as-is and then add the others once they respond.